roles/tas_single_node/tasks/certificates.yml

```diff
@@ -90,9 +85,69 @@
           set -o pipefail &&
           openssl ecparam -genkey -name prime256v1 |
           openssl ec -des3
-          -out '{{ tas_single_node_remote_fulcio_private_key }}'
+          -out '{{ tas_single_node_remote_fulcio_auto_generated_private_key }}'
           -passout 'pass:{{ tas_single_node_fulcio_ca_passphrase }}'
-        creates: "{{ tas_single_node_remote_fulcio_private_key }}"
+        creates: "{{ tas_single_node_remote_fulcio_auto_generated_private_key }}"
+      when: >
+        tas_single_node_fulcio.tas_single_node_fulcio_private_key is not defined
+        or tas_single_node_fulcio.tas_single_node_fulcio_private_key == ""
+        and (certs_dir_files.files | selectattr('path', 'equalto', tas_single_node_remote_fulcio_auto_generated_private_key) | list | length) == 0
+
+    - name: Load user-provided Fulcio private key
+      ansible.builtin.copy:
+        content: "{{ tas_single_node_fulcio.tas_single_node_fulcio_private_key }}"
+        dest: "{{ tas_single_node_remote_fulcio_user_provided_private_key }}"
+        mode: '0600'
+      when: tas_single_node_fulcio.tas_single_node_fulcio_private_key is defined and tas_single_node_fulcio.tas_single_node_fulcio_private_key != ""
+
+    - name: Clean-up old tas_single_node_remote_fulcio_user_provided_private_key
+      ansible.builtin.file:
+        path: "{{ tas_single_node_remote_fulcio_user_provided_private_key }}"
+        state: absent
+      when: tas_single_node_fulcio.tas_single_node_fulcio_private_key is not defined or tas_single_node_fulcio.tas_single_node_fulcio_private_key == ""
+
+    - name: Clean-up old tas_single_node_remote_fulcio_auto_generated_private_key
+      ansible.builtin.file:
+        path: "{{ tas_single_node_remote_fulcio_auto_generated_private_key }}"
+        state: absent
+      when: tas_single_node_fulcio.tas_single_node_fulcio_private_key is defined and tas_single_node_fulcio.tas_single_node_fulcio_private_key != ""
+
+    - name: Check if auto-generated private key exists
+      ansible.builtin.stat:
+        path: "{{ tas_single_node_remote_fulcio_auto_generated_private_key }}"
+      register: auto_generated_key_stat
+
+    - name: Check if user-provided private key exists
+      ansible.builtin.stat:
+        path: "{{ tas_single_node_remote_fulcio_user_provided_private_key }}"
+      register: user_provided_key_stat
+
+    - name: Copy auto-generated Fulcio private key into tas_single_node_remote_fulcio_private_key
+      ansible.builtin.copy:
+        src: "{{ tas_single_node_remote_fulcio_auto_generated_private_key }}"
+        dest: "{{ tas_single_node_remote_fulcio_private_key }}"
+        mode: '0600'
+        remote_src: true
+      when: auto_generated_key_stat.stat.exists
```

Comment:
I don't think we need to do these 2 stats calls. We can use the user-provided key when `tas_single_node_remote_fulcio_user_provided_private_key` is defined and non-empty, otherwise use the auto-generated key. Does that make sense?